<launch>
    <node pkg="rviz" type="rviz" name="rviz_receive2" args="-d $(find rosddsbridge)/rviz/receive2.rviz" />
    
    <!--- TF -->
    <!--node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 1.570795   0        1.570795 /map    /camera_init 10" />
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera" args="0 0 0 -1.570795 -1.570795 0        /camera /base_link   10" /-->
    <!--node pkg="tf" type="static_transform_publisher" name="odom_to_map" args="0 0 0 0 0 0 /map /odom   100" /-->
    <node pkg="tf" type="static_transform_publisher" name="velodyne_to_sensot"  args="0 0 0 0 0  0   /vehicle    /velodyne 10" />
   <!-- server own ip -->
     <node pkg="rosddsbridge" type="pcMapSub" name="pcMapSub" launch-prefix="xterm -e" respawn="true" output="screen">
		  <param name="wan_ip"  type="string"    value="192.168.1.233" />
		  <param name="port"    type="int"       value="56456" />
		  <param name="comm"    type="bool"      value="true" />
      <param name="ros_pub_PointCloud2_topic"    value= "/receive_map_cloud"/>
		  <param name="dds_sub_PointCloud2_topic"    value=   "MapPclTopic"/>
    </node>

	  <!--node pkg="rosddsbridge" type="pcOdomSub" name="pcOdomSub" launch-prefix="xterm -e" respawn="true" output="screen">
		  <param name="wan_ip"  type="string"    value="192.168.1.233" />
		  <param name="port"    type="int"       value="56458" />
		  <param name="comm"    type="bool"      value="true" />
		  <param name="ros_sub_odometry_topic"    value= "receive_odom"/>
		  <param name="dds_pub_odometry_topic"    value= "OdometryTopic"/>
    </node>

	  <node pkg="rosddsbridge" type="pcVelodyneSub" name="pcVelodyneSub" launch-prefix="xterm -e" respawn="true" output="screen">
		  <param name="wan_ip"  type="string"    value="192.168.1.233" />
		  <param name="port"    type="int"       value="56460" />
		  <param name="comm"    type="bool"      value="true" />
      <param name="ros_pub_PointCloud2_topic"    value= "/receive_velodyne_cloud"/>
		  <param name="dds_sub_PointCloud2_topic"    value=   "VelodynePclTopic"/>
    </node-->

  <node pkg="rosddsbridge" type="pcVelodyneWithOdometry" name="pcVelodyneWithOdometry" launch-prefix="xterm -e" respawn="true" output="screen">
		<param name="wan_ip"  type="string"    value="192.168.1.233" />
		<param name="port"    type="int"       value="56460" />
		<param name="comm"    type="bool"      value="true" />
        <param name="ros_pub_PointCloud2WithOdom_topic"    value= "/pclWithOdom"/>
		<param name="dds_sub_PointCloud2WithOdom_topic"    value=   "PclWithOdomTopic"/>
  </node>

    <node pkg="rosddsbridge" type="pcImageSub" name="pcImageSub" launch-prefix="xterm -e" respawn="true" output="screen">
		  <param name="wan_ip"  type="string"    value="192.168.1.233" />
		  <param name="port"    type="int"       value="56460" />
		  <param name="comm"    type="bool"      value="true" />
        <param name="ros_pub_PointCloud2WithOdom_topic"    value= "/pclWithOdom"/>
		  <param name="dds_sub_PointCloud2WithOdom_topic"    value=   "PclWithOdomTopic"/>
    </node>

    <node pkg="rosddsbridge" type="pcMeasureSub" name="pcMeasureSub" launch-prefix="xterm -e" respawn="true" output="screen">
		  <param name="wan_ip"  type="string"    value="192.168.1.233" />
		  <param name="port"    type="int"       value="56460" />
		  <param name="comm"    type="bool"      value="true" />
          <param name="ros_pub_Measure_topic"    value= "/roomVis"/>
		  <param name="dds_sub_Measure_topic"    value=   "MeasureTopic"/>
    </node>

    <!--node pkg="rosddsbridge" type="floam_laser_mapping_node" name="floam_laser_mapping_nodeDDS"  respawn="true" >
      <param name="map_resolution"    value="0.4" />
		  <param name="pubPclTopic"   value="receive_cloudMap" />
      <param name="subOdomTopic"    value= "/receive_odom"/>
		  <param name="subPclTopic"    value=   "/receive_velodyne_cloud"/>
    </node-->
    <node pkg="rosddsbridge" type="visualizationDDS" name="visualizationDDS" launch-prefix="xterm -e" respawn="true" output="screen" />
	
	<!-- second-->
	 <node pkg="rosddsbridge" type="pcVelodyneWithOdometry2" name="pcVelodyneWithOdometry2" launch-prefix="xterm -e" respawn="true" output="screen">
		<param name="wan_ip"  type="string"    value="192.168.1.233" />
		<param name="port"    type="int"       value="56460" />
		<param name="comm"    type="bool"      value="true" />
      	<param name="ros_pub_PointCloud2WithOdom_topic"    value= "/pclWithOdom2"/>
		<param name="dds_sub_PointCloud2WithOdom_topic"    value=   "PclWithOdomTopic2"/>
 	 </node>

</launch>
